### ISSUE TITLE
[Bug]: Deathrun - insta death after respawn after few minutes

### ISSUE THREAD

**Issue by rodaN90** [kus/cs2-modded-server#166]
### Server is up to date

- [X] My server is up-to-date and I'm using the latest version of this mod from this github.

### I've followed the instructions

- [X] I've followed the instructions

### Game Server Login Token

- [X] I'm using a valid Game Server Login Token or running on LAN

### Operating System

Other (please specify)

### Which mod are you playing

Other (please specify)

### Are you trying to host online or on LAN?

Online off a machine on my local network

### Metamod:Source Version

2.0.0-dev+1315

### Metamod:Source Plugins loaded

CounterStrikeSharp (v291 @ 6349c11)

### CounterStrikeSharp Version

v291 (1.0.291+6349c11)

### CounterStrikeSharp Plugins loaded

 [Client]   List of all plugins currently loaded by CounterStrikeSharp: 18 plugins loaded. [Client]   [#1:LOADED]: "Custom Votes" (1.1.3) by imi-tat0r [Client]     Allows you to create custom votes for your server. [Client]   [#2:LOADED]: "CS2 Announcement Broadcaster" (0.5.0) by Lengran [Client]     A plugin that helps server admins to broadcast announcements to users. https://github.com/lengran/CS2AnnouncementBroadcaster [Client]   [#3:LOADED]: "CS2Rcon" (1.2.0) by LordFetznschaedl [Client]     Allows for server commands to be executed from the client using !rcon [Client]   [#4:LOADED]: "CS2_ExecAfter" (1.0.0) by Kus (https://github.com/kus) [Client]     Executes a command after server event or a delay. exec_after for help [Client]   [#5:LOADED]: "GameModeManager" (1.0.50) by Striker-Nick [Client]     A simple plugin to help administrators manage custom game modes, settings, and map rotations. [Client]   [#6:LOADED]: "[Custom] Map Configurator" (V. 1.0.2 [Beta]) by Manifest @Road To Glory [Client]     Allow server owners to easily create unique configuration files on a per map basis. [Client]   [#7:LOADED]: "RockTheVote" (1.8.5) by abnerfs [Client]     https://github.com/abnerfs/cs2-rockthevote [Client]   [#8:LOADED]: "SimpleAdmin" (0.1.2) [Client]   [#9:LOADED]: "WASDMenuAPI" (1.0.5) by Interesting [Client]   [#10:UNLOADED]: "Open Prefire Prac" (0.1.41) by Lengran [Client]     A plugin for practicing prefire in CS2. https://github.com/lengran/OpenPrefirePrac [Client]   [#11:UNLOADED]: "Executes Plugin" (1.0.4) by zwolof, B3none [Client]     Community executes for CS2. [Client]   [#12:LOADED]: "Remove Map Weapons" (1.0.1) by Kus (https://github.com/kus) [Client]     Remove the weapons from the map [Client]   [#13:UNLOADED]: "Retakes Plugin" (2.0.14) by B3none [Client]     https://github.com/b3none/cs2-retakes [Client]   [#14:UNLOADED]: "Instadefuse Plugin" (2.0.0) by B3none [Client]     https://github.com/B3none/cs2-instadefuse [Client]   [#15:UNLOADED]: "Retakes Allocator Plugin" (2.3.16) by Yoni Lerner, B3none, Gold KingZ [Client]     https://github.com/yonilerner/cs2-retakes-allocator [Client]   [#16:LOADED]: "Damage Informations" (2.3.4) by K4ryuu @ KitsuneLab [Client]   [#17:UNLOADED]: "K4-Arenas" (1.5.2 (release)) by K4ryuu [Client]     An arena plugin for Counter-Strike2 [Client]   [#18:LOADED]: "SharpTimer" (0.2.6 - 5/27/2024 10:44:52 PM) by dea https://github.com/deafps/ [Client]     A CS2 Timer Plugin

### What is the bug

After changing !modes to deathrun - after a few minutes (max 5) all players die - respawn and insta die again. No way to play this mode.

**Comment by rodaN90** [comment]
Sorry, wasn't about deathrun but course.
Problem still exist. No matter what map, after a minute if you die you respawn and insta die again in a loop.

Found this as a workaround/solution:

https://github.com/kus/cs2-modded-server/pull/129

I don't know why he closed it though as it's working.

As I tried just one map and by reading his comments, it seems that all course maps have a default spawn area which after 1min triggers a death trigger... seems like those maps aren't intended to be respawned on.
By providing his spawn configs, he moves the respawn just a bit forward outside the default spawn area which isn't encased by that trigger.

I'll leave this issue open until you merge it or provide your own solution. :)

**Comment by kus** [comment]
@rodaN90 same comment as your other issue #169, I see you have a few mods unloaded which means you have switched through a few different modes before course.

Can you see if it happens when you load the server and load !course as the first game mode?

Sometimes unloading game modes isn’t 100% and weird bugs stay behind.

Just confirming you do mean course and not death run?

**Comment by rodaN90** [comment]
Yes, this is about course.
But I opened another issue #176 for deathrun as there is also a problem.

I tried with a fresh restart and the same problem happens.
Like I said, it's not a problem with the config or the plugins but the maps.

After the first minute a deathtrigger (which doesn't get deleted after first trigger - so perma active) gets loaded and all players in spawn die.
Due to the respawn of the KZ ? plugin you of course spawn in the spawn area, which still triggers the deathtrigger. So you end up dying in a loop.

The above workaround does work but needs to be manually configured for every map -> Set different respawn positions.
Also with that plugin it seems you can create start and end areas (so the timer from the KZ plugin gets started and ends to be able to create a ranking and speedrun the map)

So basically your setup works but should disable respawn/kz plugin to not die in a loop.
That way you can't respawn, can't do a timed run and have to wait for everyone to die to start a new round.
With above workaround you can have respawn and timers.

Maybe it's possible to create another menu after doing !modes->deathrun -> which then shows !1 "With Respawn/KZ/etc." and !2 "Without it"
That way people can play the maps as either !1 (fixed all that and perma fun) or !2 run the map as intended?
